rest/client.go

```diff
@@ -34,10 +34,13 @@ const (
 	httpPut    = "PUT"
 )
 
+// The type that is the function that is to be wrapped by the ClientMiddleware
 type InvocationHandler func(c *http.Client, method, url string, body interface{}, header http.Header) (*Response, error)
```

Comment:
This is a nit, but Go docstrings should start with the name of the type/function/struct/etc. https://blog.golang.org/godoc-documenting-go-code